### PR TITLE
editoast: extract all the STDCM log logic into a function

### PR DESCRIPTION
**Pull request by woshilapin**
Here is a suggestion to put everything inside a single function and extract all this weird debugging logic as far as possible of the main business logic that is in STDCM endpoint.

⚠️ Note that I targeted directly the branch of #9873

### CODE REVIEW

File: editoast/src/models/stdcm_log.rs [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+use opentelemetry::trace::TraceContextExt;
 use serde::Deserialize;
 use serde::Serialize;
+use tracing_opentelemetry::OpenTelemetrySpanExt;
 use utoipa::ToSchema;
 
 use crate::core::stdcm::Request;
 use crate::core::stdcm::Response;
+use crate::models::prelude::*;
+use crate::views::Authentication;
```
Comment (leovalais) [inline editoast/src/models/stdcm_log.rs:14]: A `views` import in `models` :/

Given how things are going, the authentication and authorization are likely to stay in `views` and in a dedicated crate. The `models` "crate" won't probably know about that layer. Wdyt about taking an `Option<i64>` instead?

File: editoast/src/models/stdcm_log.rs [hunk 2]
```diff
@@ -25,3 +30,37 @@ pub struct StdcmLog {
     pub created: DateTime<Utc>,
     pub user_id: Option<i64>,
 }
+
+impl StdcmLog {
+    // We just don't await the creation of the log entry since we want
+    // the endpoint to return as soon as possible, and because failing
+    // to persist a log entry is not a very important error here.
```
Comment (leovalais) [inline editoast/src/models/stdcm_log.rs:37]: Shouldn't this comment stay near the `spawn`? It mentions an "endpoint" and awaiting or not the return value is a decision for the caller.

**Review by leovalais — APPROVED** [review]
Great to keep the `stdcm` handler as clean as possible! There's just a few things I'd leave in that handler ⬇️

File: editoast/src/models/stdcm_log.rs (hunk 1 above)
Comment (woshilapin) [reply editoast/src/models/stdcm_log.rs:14]: Good catch, so agreed. I really quickly prototyped this solution (maybe I should have opened a draft). This is one thing to improve for sure.

File: editoast/src/models/stdcm_log.rs (hunk 2 above)
Comment (woshilapin) [reply editoast/src/models/stdcm_log.rs:37]: Good point.

**Comment by hamz2a** [comment]
As discussed, I have integrated the changes into the [PR](https://github.com/OpenRailAssociation/osrd/pull/9873), thank you very much for the suggestions. @woshilapin @leovalais

**Comment by woshilapin** [comment]
This work has been integrated into https://github.com/OpenRailAssociation/osrd/pull/9873, I'm closing it.

**Pull request closed without merging**